### PR TITLE
Helpers::detectTypes() added support for TIMESTAMPZ

### LINKED ISSUE #163: Wrong type detection in PostgreSQL
- bug report? yes
- feature request? no
- version: v2.4.1 

### Description
Data type detection does not accept all postgres supported timestamp types as "time" types. Here, postgres type "TIMESTAMP WITH TIME ZONE" is interpreted as string.

### Steps To Reproduce
Create a table with two columns "TIMESTAMP"  and "TIMESTAMP WITH TIME ZONE". Insert a record and compare the results in ActiveRow object -- the first column will be OK, the second will be string.

Fix is simple -- update Helpers.php as follows:
Changing the $typePatterns array on line with '(SMALL)?DATETIME(OFFSET)?\d*|TIME(STAMP)?' by adding .* after "STAMP".

### PR DESCRIPTION
**Pull request by WyskyNet**
…PostgreSQL [Closes #163]

(cherry picked from commit 6e4b22c)

- bug fix / new feature?   YES
- BC break? NO

Hi,
we are using latest nette/database v2.4 with postgres and there is still bug with wrong detection of timestampz data type. Bug was fixed in master, but unfortunatelly not in current stable branch...

Is there possibility to merge this cherry pick to v2.4 branch?

### CODE REVIEW

File: tests/Database/files/pgsql-nette_test3.sql [hunk 1]
```diff
@@ -10,6 +10,7 @@ CREATE TEMPORARY TABLE "types" (
   "date" date NULL,
   "time" time NULL,
   "timestamp" timestamp NULL,
+  "timestampZone" timestamp with time zone NULL,
```
Comment (dg) [inline tests/Database/files/pgsql-nette_test3.sql]: add `'timestampZone' => null` below etc.

Comment (WyskyNet) [reply tests/Database/files/pgsql-nette_test3.sql]: Hi, all tests fixed

**Comment by dg** [comment]
Thanks!

**Comment by WyskyNet** [comment]
Can you create some tag for this release, please? For example 2.4.5.1? We can use it in composer :smile:

**Comment by dg** [comment]
Done

**Pull request merged**